papr/papr2kube.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+# Convert a papr YAML file into a Kubernetes Job
+
+import os
+import sys
+import re
+import time
+import yaml
+import traceback
+import argparse
+import subprocess
+
+# XXX: switch to relative imports when we're a proper module
+from papr import PKG_DIR
+import papr.utils.parser as paprparser
+
+GH_NAME_REGEX = re.compile('^[A-Za-z0-9_.-]+$')
+
+def paprsuite2kubejob(gh_org, gh_repo, commitsha, suiteidx, suite):
+    commitsha_short = commitsha[0:10]
+    name = 'papr-{}-{}-{}-{}'.format(gh_org, gh_repo, commitsha_short, suiteidx)
+    metadata = {'name': name}
+    cmd = 'set -xeuo pipefail\n'
+    for test in suite['tests']:
+        cmd += test + '\n'
+    volumes = [
+        {'name': 'builddir', 'emptyDir': {}}
+    ]
+    containers = [{
+        'name': name,
+        'image': suite['container']['image'],
+        'volumeMounts': [
+            { 'name': 'builddir',
+              'mountPath': '/srv',
+            }
+        ],
+        'securityContext': {'runAsUser': 0},
```

Comment:
This is something we'll have to solve/workaround if we want to be able to run on most (all?) managed OCP instances though, right? I suppose we can do like Travis and provide the option between unprivileged and privileged tests, with the latter just resulting in spinning up a light VM?

Reply:
I'm sure that we can get privileges to do this in at least *some* OCP clusters, from internal OpenStack to CentOS CI.   Not to mention that at some point user namespaces are going to be plumbed through into Kube, and that will entirely change things here.